src/psyclone/tests/psyir/transformations/loop_fusion_test.py

```diff
@@ -0,0 +1,585 @@
+
+    # Test more complex loop boundaries. Note that
+    # we might actually consider simplifying these
+    # expressions
+    code = '''subroutine sub()
+              integer :: ji, jj, n
+              integer, dimension(10,10) :: s, t
+              do jj=2-1, n+1-1
+                 do ji=1, 10
+                    s(ji, jj)=t(ji, jj)+1
+                 enddo
+              enddo
+              do jj=2-1, n+1-1
+                 do ji=1, 10
+                    s(ji, jj)=t(ji, jj)+1
+                 enddo
+              enddo
+              end subroutine sub'''
+    out, _ = fuse_loops(parser, code)
+    expected = """do jj = 2 - 1, n + 1 - 1, 1
+  do ji = 1, 10, 1
+    s(ji,jj) = t(ji,jj) + 1
+  enddo
+  do ji = 1, 10, 1
+    s(ji,jj) = t(ji,jj) + 1
+  enddo
+enddo"""
+    assert expected in out
+
+
+# ----------------------------------------------------------------------------
+def test_fuse_incorrect_bounds_step(parser):
+    '''
+    Test that loop boundaries and step size must be identical.
+    '''
+
+    # Lower loop boundary
+    code = '''subroutine sub()
+              integer :: ji, jj, n
+              integer, dimension(10,10) :: s, t
+              do jj=1, n
+                 do ji=1, 10
+                    s(ji, jj)=t(ji, jj)+1
+                 enddo
+              enddo
+              do jj=2, n
+                 do ji=1, 10
+                    s(ji, jj)=t(ji, jj)+1
+                 enddo
+              enddo
+              end subroutine sub'''
+    with pytest.raises(TransformationError) as err:
+        fuse_loops(parser, code)
+    assert "Lower loop bounds must be identical, but are" in str(err.value)
+
+    # Upper loop boundary
+    code = '''subroutine sub()
+              integer :: ji, jj, n
+              integer, dimension(10,10) :: s, t
+              do jj=1, n
+                 do ji=1, 10
+                    s(ji, jj)=t(ji, jj)+1
+                 enddo
+              enddo
+              do jj=1, n+1
+                 do ji=1, 10
+                    s(ji, jj)=t(ji, jj)+1
+                 enddo
+              enddo
+              end subroutine sub'''
+    with pytest.raises(TransformationError) as err:
+        fuse_loops(parser, code)
+    assert "Upper loop bounds must be identical, but are" in str(err.value)
+
+    # Test step size
+    code = '''subroutine sub()
+              integer :: ji, jj, n
+              integer, dimension(10,10) :: s, t
+              do jj=1, n, 2
+                 do ji=1, 10
+                    s(ji, jj)=t(ji, jj)+1
+                 enddo
+              enddo
+              do jj=1, n
+                 do ji=1, 10
+                    s(ji, jj)=t(ji, jj)+1
+                 enddo
+              enddo
+              end subroutine sub'''
+    with pytest.raises(TransformationError) as err:
+        fuse_loops(parser, code)
+    assert "Step size in loops must be identical, but are" in str(err.value)
+
+    # Test step size - make sure it defaults to 1
```

Comment:
I'm not quite sure what this last bit is doing since the step size is explicitly set to '1' for both loops?

Reply:
Indeed - fixed.